tests/CardPaymentTest.php

```diff
@@ -86,11 +122,11 @@ public function testPurchaseRequest()
         $paymentObject->setCreditCard($creditCard);
 
 
-        $this->purchaseRequest = new Purchase($paymentObject);
-        $this->purchaseRequest->setClient($this->client);
+        $purchaseRequest = new Purchase($paymentObject);
+        $purchaseRequest->setClient($this->client);
```

Comment:
This feels backwards. Can't we have the interface be more like this?

```php
$client = new SwedbankPayClient();
$paymentResponse = $client->creditcard->post(/* Credit card request object */);
// The get() method returns the same as post(), so this is unnecessary, but is shown as an example:
$paymentResponse = $client->creditcard->get($paymentResponse->id);
$captureResponse = $paymentResponse->capture(/* Capture request object */);
```

Having to invoke `setClient()` all the time and not encapsulating the logic inside the different classes makes the SDK much more difficult to use.

Reply:
Yes, that's good idea. We will implement that in the next version